src/google/protobuf/compiler/csharp/csharp_message.cc

```diff
@@ -109,24 +110,62 @@ void MessageGenerator::Generate(io::Printer* printer) {
   vars["class_name"] = class_name();
   vars["access_level"] = class_access_level();
 
+  //It is critical that async class is declared first because static members must be declared BEFORE nested types
```

Comment:
Tiny nit: add a space after `//`

I'd also expand the comment... the *compiler* doesn't force you to declare static members before nested types. Is the important part the initialization order? If so, I'd say that:

```csharp
// It is critical that the async class is declared first to ensure the parser is initialized first during
// type initialization.
```